packages/authentication/test/unit/authentication.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from 'testlab';
+import {Authentication} from '../../src/authentication';
+import {BasicStrategy} from 'passport-http';
+const http = require('http');
+
+describe('Authentication', () => {
+  let passport;
+  let strategy;
+  let req;
+  let res;
+
+  describe('Basic Strategy', () => {
+    beforeEach(createBasicStrategy);
+    beforeEach(createPassport);
+    beforeEach(createMockHttpObjects);
+
+    it('authenticates successfully for correct credentials', () => {
+      const credential = 'simpson' + ':' + 'alpha';
+      const hash = new Buffer(credential).toString('base64');
+      req.headers.authorization = 'Basic ' + hash;
+      passport.authenticate(req, res, function done(err) {
+        expect(err).to.be.undefined;
```

Comment:
1) https://github.com/moll/js-must#beware-of-libraries-that-assert-on-property-access

Did you mean `to.be.undefined()`?

We should check our linter configuration, a statement like `expect(...).to.be.undefined` one should be rejected by the linter.

2) This hides the original error and its stack trace, the test reports a new assertion error pointing to this line instead. That makes debugging a test failure difficult.

The right way is to mark the test as done when an operation failed. This is typically done by calling `done(err)` (see my test above). In a synchronous test you would call `throw err`.